src/main/java/pl/orange/NextDoorBook/security/authentication/AuthController.java

```diff
@@ -0,0 +1,131 @@
+package pl.orange.NextDoorBook.security.authentication;
+
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.*;
+import pl.orange.NextDoorBook.security.jwt.JwtUtils;
+import pl.orange.NextDoorBook.security.payloads.LoginRequest;
+import pl.orange.NextDoorBook.security.payloads.MessageResponse;
+import pl.orange.NextDoorBook.security.payloads.SignupRequest;
+import pl.orange.NextDoorBook.security.payloads.UserInfoResponse;
+import pl.orange.NextDoorBook.role.ERole;
+import pl.orange.NextDoorBook.role.Role;
+import pl.orange.NextDoorBook.role.RoleRepository;
+import pl.orange.NextDoorBook.security.web_security.UserDetailsImpl;
+import pl.orange.NextDoorBook.user.User;
+import pl.orange.NextDoorBook.user.UserRepository;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+
+    @Autowired
+    AuthenticationManager authenticationManager;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    RoleRepository roleRepository;
+
+    @Autowired
+    PasswordEncoder encoder;
+
+    @Autowired
+    JwtUtils jwtUtils;
+
+    @PostMapping("/signin")
+    public ResponseEntity<?> authenticateUser(@Valid @RequestBody LoginRequest loginRequest) {
```

Comment:
skoro wcześniej pozbywaliśmy się wildcard'ów to czy w tym wypadku nie można zastąpić go typem MessageResponse ?